### PR TITLE
changes needed for clang-16 to be working

### PR DESCRIPTION
**Pull request by alexanderguzhva**

### CODE REVIEW

**Comment by liliu-z** [comment]
Hi Alex, we prefer having only one commit in each PR for better readability. Let's squash and force push this before merging. Thanks

**Comment by liliu-z** [comment]
> @alexanderguzhva 🔍 Important: PR Classification Needed!
> 
> For efficient project management and a seamless review process, it's essential to classify your PR correctly. Here's how:
> 
> 1. If you're fixing a bug, label it as kind/bug.
> 2. For small tweaks (less than 20 lines without altering any functionality), please use kind/improvement.
> 3. Significant changes that don't modify existing functionalities should be tagged as kind/enhancement.
> 4. Adjusting APIs or changing functionality? Go with kind/feature.
> 
> For any PR outside the kind/improvement category, ensure you link to the associated issue using the format: “issue: #”.
> 
> Thanks for your efforts and contribution to the community!.

@alexanderguzhva Hi Alex, plz check this. Let's put kind/improvement here with some description to help people understand this. Thanks!

**Comment by liliu-z** [comment]
/assign @Presburger 
Plz take another look at this. Thanks

**Comment by Presburger** [comment]
hi, what about boost/1.82.0, i think we use same boost version with milvus maybe better, milvus use boost/1.82.0 now.

File: conanfile.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def configure(self):
             self.options.rm_safe("fPIC")
 
     def requirements(self):
-        self.requires("boost/1.78.0")
+        self.requires("boost/1.83.0")
```
Comment (Presburger) [inline conanfile.py:84]: try boost/1.82.0?

Comment (alexanderguzhva) [reply conanfile.py:84]: boost 1.82.0 works as well. Alternatively, milvus can go to 1.83.0. Personally, I see no reason to stay with 1.82.0 if 1.83.0 is already out

**Review by Presburger — APPROVED** [review]

**Pull request merged**